backtrace-library/src/main/java/backtraceio/library/breadcrumbs/BacktraceBreadcrumbs.java

```diff
@@ -0,0 +1,342 @@
+                backtraceBroadcastReceiver.getIntentFilter(breadcrumbTypesToEnable));
+
+        if (breadcrumbTypesToEnable.contains(BacktraceBreadcrumbType.SYSTEM)) {
+            backtraceComponentListener = new BacktraceComponentListener(this);
+            context.registerComponentCallbacks(backtraceComponentListener);
+
+            backtraceActivityLifecycleListener = new BacktraceActivityLifecycleListener(this);
+            if (context instanceof Application) {
+                ((Application) context).registerActivityLifecycleCallbacks(backtraceActivityLifecycleListener);
+            }
+        }
+    }
+
+    public boolean enableBreadcrumbs(Set<BacktraceBreadcrumbType> enabledBreadcrumbTypes,
+                                  boolean isNewInstance,
+                                  BacktraceBase staleBreadcrumbLogSender) {
+        if (isNewInstance == true) {
+            handleExistingBreadcrumbLogFile(context, staleBreadcrumbLogSender);
+        }
+        if (backtraceBreadcrumbsLogger == null) {
+            try {
+                backtraceBreadcrumbsLogger = new BacktraceBreadcrumbsLogger(getBreadcrumbLogPath(this.context), this.maxBreadcrumbLogSizeBytes);
+            } catch (Exception ex) {
+                BacktraceLogger.e(LOG_TAG, "Could not start the Breadcrumb logger due to: " + ex.getMessage());
+                return false;
+            }
+        }
+
+        registerAutomaticBreadcrumbReceivers(enabledBreadcrumbTypes);
+
+        this.enabledBreadcrumbTypes = enabledBreadcrumbTypes;
+
+        // We should log all breadcrumb configuration changes in the breadcrumbs
+        addConfigurationBreadcrumb();
+        return true;
+    }
+
+    public boolean enableBreadcrumbs(boolean isNewInstance, BacktraceBase staleBreadcrumbLogSender) {
+        final Set<BacktraceBreadcrumbType> enabledBreadcrumbTypes = new HashSet<BacktraceBreadcrumbType>(){{
+            add(BacktraceBreadcrumbType.CONFIGURATION);
+            add(BacktraceBreadcrumbType.HTTP);
+            add(BacktraceBreadcrumbType.LOG);
+            add(BacktraceBreadcrumbType.MANUAL);
+            add(BacktraceBreadcrumbType.NAVIGATION);
+            add(BacktraceBreadcrumbType.SYSTEM);
+            add(BacktraceBreadcrumbType.USER);
+        }};
+
+        return enableBreadcrumbs(enabledBreadcrumbTypes, isNewInstance, staleBreadcrumbLogSender);
+    }
+
+    private void unregisterAutomaticBreadcrumbReceivers() {
+        this.context.unregisterReceiver(backtraceBroadcastReceiver);
+        backtraceBroadcastReceiver = null;
+
+        this.context.unregisterComponentCallbacks(backtraceComponentListener);
+        backtraceComponentListener = null;
+
+        if (context instanceof Application) {
+            ((Application) context).unregisterActivityLifecycleCallbacks(backtraceActivityLifecycleListener);
+        }
+    }
+
+    public void disableBreadcrumbs() {
+        if (this.isBreadcrumbsEnabled()) {
```

Comment:
I guess the user might want to disable our automatic breadcrumb receivers at certain points. Really `disableBreadcrumbs` is  just unregistering all automatic breadcrumb receivers.

Reply:
Yeah, but the only reason why I think user might want to disable breadcrumbs receiver and will use this api is to prevent breadcrumbs message on application exit or something like that. Is it worth to expose this api at all? Can we list those use cases here?

Reply:
I think you're right. Disable adds complexity and isn't useful. Even if user wants to change the breadcrumb types at runtime, they can call `enableBreadcrumbs` again with the new types they want to enable.

Reply:
Removed `disableBreadcrumbs`, thank you!